lib/jsonapi/plugs/query_parser.ex

```diff
@@ -111,8 +113,9 @@ defmodule JSONAPI.QueryParser do
     end
   end
 
-  def parse_fields(config, map) when map_size(map) == 0, do: config
+  def parse_fields(%Config{} = config, fields) when fields == %{}, do: config
 
+  @spec parse_fields(Config.t(), map()) :: map() | no_return()
```

Comment:
would this `@spec` go above the 1st parse_fields/2 method?

Reply:
Oops! Fixed.